L1Trigger/L1TGlobal/src/ZdcEnergySumCondition.cc

```diff
@@ -0,0 +1,181 @@
+  // number of trigger objects in the condition: there is only one object
+  int iCondition = 0;
+
+  // condition result condResult set to true if the energy sum
+  // passes all requirements
+  bool condResult = false;
+
+  // store the indices of the calorimeter objects
+  // from the combination evaluated in the condition
+  SingleCombInCond objectsInComb;
+
+  // clear the m_combinationsInCond vector
+  (combinationsInCond()).clear();
+
+  // clear the indices in the combination
+  objectsInComb.clear();
+
+  const BXVector<const l1t::EtSum*>* candVecZdc = m_uGtB->getCandL1ZdcEtSum();
+
+  // Look at objects in bx = bx + relativeBx
+  int useBx = bxEval + m_gtZdcEnergySumTemplate->condRelativeBx();
+
+  // Fail condition if attempting to get Bx outside of range
+  if ((useBx < candVecZdc->getFirstBX()) || (useBx > candVecZdc->getLastBX())) {
+    return false;
+  }
+
+  // If no candidates, no use looking any further
+  int numberObjectsZdc = candVecZdc->size(useBx);
+
+  if (numberObjectsZdc < 1) {
+    return false;
+  }
+
+  const ZdcEnergySumTemplate::ObjectParameter objPar = (*(m_gtZdcEnergySumTemplate->objectParameter()))[iCondition];
+  bool condGEqVal = m_gtZdcEnergySumTemplate->condGEq();
+
+  l1t::EtSum candZdcPlus;
+  l1t::EtSum candZdcMinus;
+  unsigned int candZDCPEsum = 0;
+  unsigned int candZDCMEsum = 0;
+  bool myres = false;
+
+  for (int iEtSum = 0; iEtSum < numberObjectsZdc; ++iEtSum) {
+    l1t::EtSum candZdc = *(candVecZdc->at(useBx, iEtSum));
+
+    if (l1t::EtSum::EtSumType::kZDCP == candZdc.getType())
+      candZdcPlus = *(candVecZdc->at(useBx, iEtSum));
+    else if (l1t::EtSum::EtSumType::kZDCM == candZdc.getType())
+      candZdcMinus = *(candVecZdc->at(useBx, iEtSum));
+    LogDebug("L1TGlobal") << "CANDZdc: " << candZdc.hwPt() << ", " << useBx << ", " << candZdc.getType() << std::endl;
+
+    if (candZdc.getType() == l1t::EtSum::EtSumType::kZDCP) {
+      candZDCPEsum = candZdcPlus.hwPt();
+      myres = checkThreshold(objPar.etLowThreshold, objPar.etHighThreshold, candZDCPEsum, condGEqVal);
+    } else if (candZdc.getType() == l1t::EtSum::EtSumType::kZDCM) {
+      candZDCMEsum = candZdcMinus.hwPt();
+      myres = checkThreshold(objPar.etLowThreshold, objPar.etHighThreshold, candZDCMEsum, condGEqVal);
+    }
+
+    else {
+      LogDebug("L1TGlobal") << "\t\t l1t::EtSum failed ZDC checkThreshold" << std::endl;
+      return false;
+    }
+
+    LogDebug("L1TGlobal") << "ZDC EtSumType object from ZdcEnergySumTemplate: "
+                          << "\n objPar.etLowThreshold = " << objPar.etLowThreshold
+                          << "\n objPar.etHighThreshold = " << objPar.etHighThreshold
+                          << "\n candZDCPEsum = " << candZDCPEsum << "\n candZDCMEsum = " << candZDCMEsum
+                          << "\n condGEqVal = " << condGEqVal << "\n myres = " << myres << std::endl;
+  }
+
+  if (!(myres && condGEqVal))
```

Comment:
According to `CondFormats/L1TObjects/interface/L1GtCondition.h`, `condGEqVal` indicates
```c++
/// the operator used for the condition (>=, =): true for >=
```

Is the meaning different for ZDC energy sums?

If the meaning is the usual one, why do you check
```c++
if (!(myres && condGEqVal))
```
instead of simply
```c++
if (not myres)
```
?

Reply:
You are right Andrea, it is used in the `checkThreshold` function similarly to all the other occurrencies.
I somehow misinterpreted the usage. Thank you.

Reply:
@elfontan fixes for above can be found in the latest commit